sourceCode.py

```diff
@@ -1 +1,523 @@
+
+##########################################################
+##                                                      ##
+## grammar lesson 1: first person becomes second person ##
+##                                                      ##
+##########################################################
+
+gReflections = {
+    "am"   : "are",
+    "was"  : "were",
+    "i"    : "you",
+    "i'd"  : "you would",
+    "i've"  : "you have",
+    "i'll"  : "you will",
+    "my"  : "your",
+    "are"  : "am",
+    "you've": "I have",
+    "you'll": "I will",
+    "your"  : "my",
+    "yours"  : "mine",
+    "you"  : "me",
+    "me"  : "you"
+    }
+  
+                                ########################################################################################  
+                                ##   -------------------- Topic Functions and Libraries -------------------------     ##
+                                ########################################################################################
+
+
+
+##############################
+##                          ##
+## small talk with Jordan   ##
+##                          ##
+##############################
+
+## create the library for small-talk with Jordan
+library_smalltalk = [
+    [r'Hi(.*)',
+     [  "Hi, how are you?",
+      "Hey, whazzup?",
+      "Hello you!"]],
+    
+    [r'How are you?',
+     [  "I am fine, thanks. You?",
+      "Great, you?",
+      "I am doing wonderful. How are you doing?"]],
+    
+    [r'I am (.*)',
+     [  "That is nice to hear!",
+      "How wonderful."]],
+    
+    [r'I want (.*)',
+     [  "Yeah, me too!",
+      "Why do you want %1?",
+      "What would you do if you got %1?",
+      "If you got %1, then what would you do?"]],
+    
+    [r'It is (.*)',
+     [  "You seem very certain.",
+      "Interesting. Why?"]],
+    
+    [r'Because (.*)',
+     [  "I see."]],
+    
+    [r'(.*) weather(.*)',
+     [  "I love talking about the weather.",
+      "Does it seem strange to talk to a computer about the weather?",
+      "It is super sunny today, right?"]],
+    
+    [r'quit',
+     [  "Thank you for talking with me.",
+      "Good-bye!.",
+      "Have a nice day!"]],
+    ]
+    
+## create the function to have some small-talk with Jordan
+
+import random
+import re
+
+def smalltalk_Jordan(userInput):
+    keys_smalltalk= list(map(lambda x:re.compile(x[0], re.IGNORECASE), library_smalltalk))
+    values_smalltalk = list(map(lambda x:x[1],library_smalltalk))
+    
+    while True:
+        userInput = input(">>> ")
+        for i in range(0, len(keys_smalltalk)):
+            match = keys_smalltalk[i].match(userInput)
+            if match:
+                resp = random.choice(values_smalltalk[i])
+                pos = resp.find('%')
+                print(resp)
+            else:
+                print("I did not understand what you said")
+    
```

Comment:
You probably ran this function yourself already but I just wanted to tell you in case you missed it or forgot: it prints "I did not understand what you said" 8 times in case it doesn't match anything or even 7 times in case there is a match (and I think in that case you actually only want it to print that once overall). I tried this code below and added some comments for further improvement :)
```suggestion
def smalltalk_Jordan():
    # I'm not sure if it is good practice to leave the input empty but the userInput is not necessary as an argument
    keys_smalltalk = list(map(lambda x:re.compile(x[0], re.IGNORECASE), library_smalltalk))
    values_smalltalk = list(map(lambda x:x[1],library_smalltalk))
    
    while True:
        # I think it would be nice (but not necessary) to have something like while user_wants_to_quit == False:
        userInput = input(">>> ")
        resp = "I did not understand what you said" # initalize this
        counter = 0  # initalize this
        while resp == "I did not understand what you said" and counter < len(keys_smalltalk):
            for i in range(0, len(keys_smalltalk)):
                match = keys_smalltalk[i].match(userInput)
                if match:
                    resp = random.choice(values_smalltalk[i])
                    pos = resp.find('%') # Maybe you already saw this but this is not working
                counter += 1
                    
        print(resp)
```

Reply:
Maybe it would make sense to first check whether there is a match and then print the response. So, only print 'I don't understand ...' when match == False. Additionally, what are you going to do when there are multiple matches. For example, in the Smalltalk function, if I type 'because weather' it will respond with two responses. Maybe you could include something that you randomly choose one of the responses.